### PR TITLE
[api] item with authorities

### PR DESCRIPTION
**Pull request by rastislav-chynoransky**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes https://jira.sng.sk/browse/WEBUMENIA-1868

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [x] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by eronisko — CHANGES_REQUESTED** [review]
I'm marking this as 'request changes', but I mostly want to open a discussion.

1. I don't think we need to use Elasticsearch for this because we're not doing any searches. I'd guess that most of the `detail` method, could be replaced with something like

    ```php
    Item::findOrFail($id)->with('authorities');
    ```
    which I would prefer because it saves us from using 2 different libraries and adding a new trait just to run the query & adding new fields to the search index that we won't use outside of the API.

    I guess there's a legacy from when the ES *was* the API, but I think we can move away from that now and have a solution much closer to what we actually need.

    And yes, we might use the Scout functionality for `Item`s later on, but I don't think this API endpoint is the right place for search. (Unless I've missed something -- like the `$q` param)

1.  Not really part of this PR, but what I miss here very much is some kind of explicit contract or expectation of what exactly the API will return. The reason why I like the [Laravel API Resources](https://laravel.com/docs/9.x/eloquent-resources#concept-overview) is that you're encouraged to explicitly list each attribute you want to expose. 

     I think the 'opt-in' kind of approach is useful for maintaining APIs, because you know exactly what you're exposing and what to look for when renaming/re-factoring. Implicitly exposed attributes can be consumed by a service and changing it would not trigger any alarms (e.g. test failures) but it could easily break downstream systems. The question "why is this exposed on API" can be usually answered by looking at Git history, even if you don't have any API documentation.

    So I think it would be useful to use an explicit list of exposed fields, preferably in the form of a Laravel API Resource linked above.

**Comment by rastislav-chynoransky** [comment]
@eronisko I agree with you. These changes will require a bit more time since there are already multiple consumers of this API. So I'd say it's a question for @mcudrnak whether the release of this feature for kod.sng.sk could be postponed a little.

**Review by eronisko — COMMENTED** [review]
This looks so much better, thank you. Please see my comments about the wrapping setting, I'd suggest to keep it on. cc @igor-kamil

File: app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -55,6 +57,8 @@ public function register()
     public function boot()
     {
         Paginator::useBootstrap();
+        AuthorityResource::withoutWrapping();
+        ItemResource::withoutWrapping();
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app/Providers/AppServiceProvider.php:61]: I was kinda curious what you'd say about these. The data wrapping always seemed redundant to me. And `withoutWrapping` has no effect on paginated collections: https://laravel.com/docs/9.x/eloquent-resources#data-wrapping-and-pagination But I'm ok with removing those.

File: routes/api.php [hunk 2]
```diff
@@ -23,7 +24,11 @@
...
-    Route::get('items', [ItemController::class, 'index']);
-    Route::get('items/aggregations', [ItemController::class, 'aggregations']);
-    Route::get('items/{id}', [ItemController::class, 'detail']);
+    Route::get('items', [V1ItemController::class, 'index']);
+    Route::get('items/aggregations', [V1ItemController::class, 'aggregations']);
+    Route::get('items/{id}', [V1ItemController::class, 'detail']);
+});
+
+Route::prefix('v2')->group(function () {
+    Route::get('items/{id}', [V2ItemController::class, 'detail']);
```
Comment (eronisko) [inline routes/api.php:33]:
```suggestion
    Route::get('items/{id}', [V2ItemController::class, 'show']);
```

File: app/Providers/AppServiceProvider.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (eronisko) [reply app/Providers/AppServiceProvider.php]: Thanks. Yeah they seem redundant and they'll have to be dealt with on the client side, but I believe they'll help us maintain a more stable interface.

**Review by eronisko — APPROVED** [review]
LGTM, thank you for the changes. As a public interface, the API should probably also include some basic tests, but I'm fine with this going out as-is. Thanks again.

**Comment by rastislav-chynoransky** [comment]
> LGTM, thank you for the changes. As a public interface, the API should probably also include some basic tests, but I'm fine with this going out as-is. Thanks again.

Thanks for your thorough review!

**Pull request merged**